### PR TITLE
1744 by Inlead: Refactor ting_search_carousel

### PR DESCRIPTION
**Pull request by makcumm**
Når jeg klikker "Fjern", så AJAX reloader den, men den har ikke fjernet tab'en men kun fjernet titlen (hvilket er nemt at overse). Man skal gemme før den forsvinder helt.

Formularen bruger heller ikke Drupals add-more functionalitet, og vægtningen er også underlig: http://platform.dandigbib.org/issues/269

Den trænger til en kærlig hånd. Man kunne evt. slå det sammen med at gøre det muligt at tilføje flere karruseller.

### CODE REVIEW

**Comment by AnatolyMuntean** [comment]
I don't have something to complain about in those particular commits, however the codebase of the admin logic is obsolete and is hard to maintain. The most concerning part is the ajax form for queries manipulation - better to rework it with proper approach that Drupal ajax forms provide.

File: modules/ting_search_carousel/ting_search_carousel.admin.inc [hunk 1]
```diff
@@ -276,24 +276,26 @@ function ting_search_carousel_query_form($item = array(), $index = 0) {
...
 
-  $form['data']['remove'] = array(
-    '#type' => 'submit',
-    '#value' => t('Remove'),
-    '#name' => 'ting_search_carousel_remove_' . $index,
-    '#search_id' => $index,
-    '#attributes' => array(
-      'class' => array(
-        'remove',
+  if ($index != 0) {
```
Comment (avoykov) [inline modules/ting_search_carousel/ting_search_carousel.admin.inc]: From   if (empty($form_state['num_searches'])) { to   if (!isset($form_state['num_searches'])) {

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: c5c177c865a751a728b88617cccf11dcd391afe4
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: c5c177c865a751a728b88617cccf11dcd391afe4
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

File: modules/ting_search_carousel/ting_search_carousel.admin.inc [hunk 2]
```diff
@@ -84,116 +84,104 @@ function ting_search_carousel_settings_admin_form($form, &$form_state) {
...
-    }
+  if (empty($form_state['num_searches'])) {
+    $form_state['existing_searches'] = variable_get('ting_search_carousel_searches', array());
   }
 
-  // Whether add a new query field.
-  if ($clicked_btn == 'add_query') {
-    $searches_num++;
-  }
+  $form['ting_search_carousel']['ting_searches'] = array(
```
Comment (xendk) [inline modules/ting_search_carousel/ting_search_carousel.admin.inc]: Call this `$form['ting_search_carousel_searches']` and move the `#tree => TRUE` and wrapper markup from the form here. That'll make `system_settings_form_submit()` save directly to the variable.

File: modules/ting_search_carousel/ting_search_carousel.admin.inc [hunk 3]
```diff
@@ -84,116 +84,104 @@ function ting_search_carousel_settings_admin_form($form, &$form_state) {
...
-    $unsubmitted = array();
-    // Maybe prev field had some data in it...
-    if (isset($form_state['input']['title'][$index])) {
-      $unsubmitted = array(
-        'title' => $form_state['input']['title'][$index],
-        'subtitle' => $form_state['input']['subtitle'][$index],
-        'query' => $form_state['input']['query'][$index],
-      );
+  if (!isset($form_state['num_searches'])) {
+    $ids = array_keys($form_state['existing_searches']);
```
Comment (xendk) [inline modules/ting_search_carousel/ting_search_carousel.admin.inc]: Why bother with both `existing_searches` and `num_searches`? Just keep one which is the current state, and preload it with `variable_get('ting_search_carousel_searches', array())`;

File: modules/ting_search_carousel/ting_search_carousel.admin.inc [hunk 4]
```diff
@@ -256,65 +238,67 @@ function theme_ting_search_carousel_admin_form(array $variables) {
...
+    '#title_display' => 'invisible',
   );
 
-  $form['title[]'] = array(
+  $form['data']['title'] = array(
     '#type' => 'textfield',
     '#title' => t('Title'),
     '#description' => t('The title of the search is used for the tab in the carousel. Keep it short.'),
     '#default_value' => isset($item['title']) ? $item['title'] : '',
     '#prefix' => '<div class="search-carousel-query">',
```
Comment (xendk) [inline modules/ting_search_carousel/ting_search_carousel.admin.inc]: Remove this.

File: modules/ting_search_carousel/ting_search_carousel.admin.inc [hunk 5]
```diff
@@ -256,65 +238,67 @@ function theme_ting_search_carousel_admin_form(array $variables) {
...
-    '#attributes' => array(
-      'class' => array(
-        'sort-hidden',
-      ),
-    ),
+    '#title_display' => 'invisible',
   );
 
-  $form['title[]'] = array(
+  $form['data']['title'] = array(
```
Comment (xendk) [inline modules/ting_search_carousel/ting_search_carousel.admin.inc]: Remove the `['data']` and render the individual elements in the table. It'll mess with the `#tree`.

File: modules/ting_search_carousel/ting_search_carousel.module [hunk 6]
```diff
@@ -105,7 +105,7 @@ function ting_search_carousel_block_info() {
 function ting_search_carousel_block_view($delta) {
   return array(
     'content' => theme('ting_search_carousel', array(
-        'searches' => variable_get('ting_carousel_search_queries', array()),
```
Comment (xendk) [inline modules/ting_search_carousel/ting_search_carousel.module]: This needs an update hook to migrate old variable to the new.

**Comment by JrgenGNielsen** [comment]
Please fix code according to Thomas' comments, and rebase code to fix merge conflicts

**Comment by cableman** [comment]
This PR should not be used as this https://platform.dandigbib.org/issues/1770 introduces ting_search_carousel based on slick.

See #826

**Comment by cableman** [comment]
Please node that rename "ting_carousel_search_queries" will break https://github.com/easySuite/ding_redia_variables/blob/development/ding_redia_variables.module#L50 which is used by some libraries mobile apps

**Comment by guddo** [comment]
I think this can be closed in favour of #1007.

**Comment by DDBdrift** [comment]
https://platform.dandigbib.org/issues/269 | Closed

**Pull request closed without merging**